# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/generic/nsFrame.cpp
# Commit: 96da9d241051
# Full Hash: 96da9d241051d222bdd693bdb519d058a013b59c
# Author: Doug Thayer <dothayer@mozilla.com>
# Date: 2019-03-23 09:46:24
# Regressor Bug: 1441308
# File Overlap Count: 2
# Description:
#   Bug 1441308 - Core renderroot splitting changes r=kats,sotaro
#   
#   This is a large patch that contains all of the core changes for
#   renderroot splitting.
#   
# ==============================================================================

diff -r 8cefa694f811 -r 96da9d241051 layout/generic/nsFrame.cpp
--- a/layout/generic/nsFrame.cpp	Fri Mar 22 18:28:31 2019 +0000
+++ b/layout/generic/nsFrame.cpp	Fri Mar 22 18:28:42 2019 +0000
@@ -10551,8 +10551,19 @@
 void nsIFrame::CreateOwnLayerIfNeeded(nsDisplayListBuilder* aBuilder,
                                       nsDisplayList* aList,
                                       bool* aCreatedContainerItem) {
-  if (GetContent() && GetContent()->IsXULElement() &&
-      GetContent()->AsElement()->HasAttr(kNameSpaceID_None, nsGkAtoms::layer)) {
+  wr::RenderRoot renderRoot = gfxUtils::GetRenderRootForFrame(this)
+      .valueOr(wr::RenderRoot::Default);
+
+  if (renderRoot != wr::RenderRoot::Default) {
+    aList->AppendToTop(MakeDisplayItem<nsDisplayRenderRoot>(
+        aBuilder, this, aList, aBuilder->CurrentActiveScrolledRoot(),
+        renderRoot));
+    if (aCreatedContainerItem) {
+      *aCreatedContainerItem = true;
+    }
+  } else if (GetContent() && GetContent()->IsXULElement() &&
+             GetContent()->AsElement()->HasAttr(kNameSpaceID_None,
+                                                nsGkAtoms::layer)) {
     aList->AppendToTop(MakeDisplayItem<nsDisplayOwnLayer>(
         aBuilder, this, aList, aBuilder->CurrentActiveScrolledRoot()));
     if (aCreatedContainerItem) {